model/service/StructuredVariablesToItemStateService.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ */
+
+declare(strict_types=1);
+
+namespace oat\ltiOutcomeUi\model\service;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\taoOutcomeUi\helper\ResponseVariableFormatter;
+use taoResultServer_models_classes_ResponseVariable as ResponseVariable;
+
+class StructuredVariablesToItemStateService extends ConfigurableService
```

Comment:
You should use DI instead of configurable service

Reply:
I didn't want to set up the DI just for this solution, because at the end a backport is created for it, but I will do it

Reply:
Do you have DI available in backported version?

Reply:
I do not know